sled-agent/src/instance.rs

```diff
@@ -867,64 +985,228 @@ impl InstanceRunner {
                             port.name().into(),
                             port.slot().into(),
                         ),
+                    ))?;
+
+                let nic = &self.requested_nics[pos];
```

Comment:
i know this is code motion, but the `iter/position/map(nics[pos])` could be `iter/find(nic.slot == port.slot())` without the index operation here (`pos` is otherwise unused and the `[]` made me think about panic safety...)